clang/lib/CIR/Lowering/ThroughMLIR/LowerCIRLoopToSCF.cpp

```diff
@@ -293,6 +352,10 @@ class CIRForOpLowering : public mlir::OpConversionPattern<cir::ForOp> {
                   mlir::ConversionPatternRewriter &rewriter) const override {
     SCFLoop loop(op, &rewriter);
     loop.analysis();
+    if (!loop.isCanonical()) {
+      mlir::emitError(op.getLoc(), "cannot handle non-canonicalized loop");
```

Comment:
You can also do `return mlir::emitError(op.getLoc(), "cannot handle non-canonicalized loop");`